README.md

```diff
@@ -26,13 +26,39 @@ OpenSearch Continous Integration is an open source CI system for OpenSearch and
 - Deploy stacks with `npm run cdk deploy`
 
 ## Deployment
+
+### dev deployment 
 1. Setup your local machine to credentials to deploy to the AWS Account
-2. Deploy the ci config stack, `npm run cdk deploy CI-Config-Dev`, takes ~1 minute to deploy
-3. [Optional] Configure the elements of the config stack for SSL configuration
-4. Deploy the ci stack, without ssl, `npm run cdk deploy CI-Dev -- --parameters useSsl=false`, takes ~10 minutes to deploy
-5. Log onto the AWS Console of the account, navigate to [cloud watch](https://console.aws.amazon.com/cloudwatch/home), open log groups, looking for `JenkinsMainNode/var/log/jenkins/jenkins.log`
-6. Search the logs for `Jenkins initial setup is required. An admin user has been created and a password generated.` After that entry the password for the jenkins instance will be in the cloudwatch logs.
-7. Go to the `CI-Dev.JenkinsExternalLoadBalancerDns` url returned by CDK output to access the jenkins host.
+2. Deploy the ci-config-stack, `npm run cdk deploy CI-Config-Dev`, takes ~1 minute to deploy
+3. [Optional](#ssl-configuration) Configure the elements of the config stack for SSL configuration `npm run cdk deploy CI-Dev -- --parameters useSsl=true`
+4. [Optional](#setup-oidc-via-federate) Configure the elements setting up oidc via federate `npm run cdk deploy CI-Dev -- --parameters devmode=false`
+5. Deploy the ci-stack, without ssl, takes ~10 minutes to deploy
+6. Log onto the AWS Console of the account, navigate to [cloud watch](https://console.aws.amazon.com/cloudwatch/home), open log groups, looking for `JenkinsMainNode/var/log/jenkins/jenkins.log`
+7. Search the logs for `Jenkins initial setup is required. An admin user has been created and a password generated.` After that entry the password for the jenkins instance will be in the cloudwatch logs.
+8. Go to the `CI-Dev.JenkinsExternalLoadBalancerDns` url returned by CDK output to access the jenkins host.
+
+### Executing Optional Tasks
+#### SSL Configuration
+1. Locate the secret manager arns in the ci-config-stack outputs
+2. Update the secret value for the `certContentsSecret` with the certificate contents
+3. Update the secret value for the `privateKeySecret` with the certificate private key
+4. Upload the certificate to IAM. See [docs](https://docs.aws.amazon.com/cli/latest/reference/iam/upload-server-certificate.html)
+5. Update the secret value for the `certificateArnSecret` with the certificate arn generated by IAM
+6. Run with paramenter `npm run cdk deploy CI-Dev -- --parameters useSsl=true`
+7. Continue with [next steps](#dev-deployment)
+
+#### setup OIDC via Federate
```

Comment:
Spell out OIDC in documentation, or if you are going to use the initialism also include the full spelling next to it for folks that haven't used it before.